src/layouts/AppLayout/index.tsx

```diff
@@ -211,6 +234,44 @@ const AppLayout: FunctionComponent<AppLayoutProps> = ({
         setNotificationsBoxHeight(notificationsBoxRef.current?.offsetHeight || 0);
     }, [notificationsBoxRef, notifications]);
 
+    useEffect(() => {
+        if (notificationsProp) {
+            setNotifications((prevNotifications) => [
+                ...prevNotifications,
+                ...notificationsProp.filter((np) => !prevNotifications.find((pn) => pn.id === np.id)),
+            ]);
+        }
+    }, [notificationsProp]);
+
+    const handleDismissNotification = useCallback(
+        (id?: string) => {
+            if (id) {
+                setNotifications((prevNotifications) => prevNotifications.filter((n) => n.id !== id));
+            } else {
+                setNotifications([]);
+            }
+        },
+        [setNotifications]
+    );
+
+    const handleAddNotification = useCallback(
+        (newNotification: Notification) => {
+            setNotifications((prevNotificsations) =>
+                [
+                    {
+                        ...newNotification,
+                        onDismiss: () => {
+                            newNotification.onDismiss?.();
+                            handleDismissNotification(newNotification.id);
+                        },
+                    },
+                    ...prevNotificsations,
+                ].slice(0, maxNotifications)
```

Comment:
Should we call the sliced notifications `onDismiss` before slice the notifications array?

Reply:
Good point